include/Gaffer/Context.h

```diff
@@ -242,7 +242,7 @@ class GAFFER_API Context : public IECore::RefCounted
 				EditableScope( const ThreadState &threadState );
 				~EditableScope();
 
-				template<typename T>
+				template<typename T, typename = typename std::enable_if< std::is_base_of<IECore::Data, typename std::remove_pointer<T>::type >::value || !std::is_pointer<T>::value >::type >
```

Comment:
I _think_ this is breaking compatibility with previous versions, because it is adding a template argument. Is there any way we can disambiguate things purely with the enabler on the new method?

Reply:
No, I don't think that could be possible.  As long as the old version of set() matches any `const T &value`, that will try to compile for any call that tries to set a pointer.

What kind of compatibility are you worried about ... binary compatibility, because this isn't labelled as inline?  Doesn't a template have to be compiled by the client anyway?

I guess the alternative would be moving the switch between old vs new inside of one set function, where it dispatched using some sort of internal mechanism.  This would allow hiding the API change, but it would be a bit annoying to set up ( one of those places where we need `constexpr if` )

Reply:
I guess my worry mostly stems from not having a good understanding of the implications here. What you say about everything having to have been compiled by the client already does make sense, but I'm honestly out of my depth, and it doesn't seem unimaginable that there's some unforeseen consequence waiting to bite us. I guess you might be the best to test this in practice, by running a compiled-with-the-last-release IERendering inside a build from this PR?

Reply:
Quick note on what we discussed yesterday : if we use `enable_if` on the return type, does that mean the name-mangling for the function isn't affected?

Reply:
Putting it in the return type seems fine.

Reply:
Did a bit of testing with this, and couldn't find any `EditableScope::set` symbols exported from any of the Gaffer libraries. So the choice of enabler may have been moot anyway.